File: nan21.dnet.module.bd/nan21.dnet.module.bd.ui.extjs/src/main/resources/ds/EngineLovDs.js
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
Ext.define("net.nan21.dnet.module.bd.elem.ds.model.EngineLovDs" ,{
	extend: 'Ext.data.Model',
	fields: [
		{name:"active", type:"boolean"},
		{name:"clientId", type:"int", useNull:true},
		{name:"description", type:"string"},
		{name:"id", type:"int", useNull:true},
		{name:"name", type:"string"},
		{name:"type", type:"string"}
	]});
Ext.define("net.nan21.dnet.module.bd.elem.ds.model.EngineLovDsFilter" ,{
	extend: 'Ext.data.Model',
	fields: [
		{name:"active", type:"boolean", useNull:true},
		{name:"description", type:"string"},
		{name:"id", type:"int", useNull:true},
		{name:"name", type:"string"},
		{name:"type", type:"string"}
	]
});
Ext.define("net.nan21.dnet.module.bd.elem.ds.param.EngineLovDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
});

File: nan21.dnet.module.bd/nan21.dnet.module.bd.presenter/src/main/java/net/nan21/dnet/module/bd/elem/ds/model/EngineLovDs.java
/* 
 * DNet eBusiness Suite
 * Copyright: 2010-2013 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.bd.elem.ds.model;

import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;
import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.presenter.model.base.AbstractTypeLov;
import net.nan21.dnet.module.bd.elem.domain.entity.Engine;

@Ds(entity = Engine.class, jpqlWhere = " e.active = true ", sort = {@SortField(field = EngineLovDs.f_name)})
public class EngineLovDs extends AbstractTypeLov<Engine> {

	public static final String f_type = "type";

	@DsField
	private String type;

	public EngineLovDs() {
		super();
	}

	public EngineLovDs(Engine e) {
		super(e);
	}

	public String getType() {
		return this.type;
	}

	public void setType(String type) {
		this.type = type;
	}
}
